pkg/popeye.go

```diff
@@ -256,6 +257,8 @@ func (p *Popeye) lint() (int, int, error) {
 			continue
 		}
 
+		ctx = context.WithValue(ctx, internal.KeyNamespace, orig_ns)
```

Comment:
Context is immutable. This call will create a brand new context every time. 
I think it's best to only alter the original context if we have a clusterwide resource.

Reply:
Thanks - being new to Go in general I hope I am not heading off in the wrong direction but if I test conditionally within the loop to see if the namespace has been reset in ctx, and only if it has, mutate it using a function like I have included here seems to work. Was this what you had in mind?